core/src/main/java/org/frankframework/pipes/ReplacingVariablesInputStream.java

```diff
@@ -0,0 +1,176 @@
+/*
+   Copyright 2017 Simon Haoran Liang, 2019-2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.pipes;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.stream.IntStream;
+
+/**
+ * Copyright 2019-2024 WeAreFrank!
+ * Based on original concept created by simon on 8/29/17.
+ * Copyright 2017 Simon Haoran Liang
+ * <a href="https://gist.github.com/lhr0909/e6ac2d6dd6752871eb57c4b083799947">...</a>
+ * <p>
+ * Replaces variable placeholders with values from the given keyValuePairs in a InputStream implementation.
+ *
+ * @author Erik van Dongen
+ */
+public class ReplacingVariablesInputStream extends InputStream {
+
+	private static final byte CLOSING_CURLY_BRACE = 125;
+	private static final byte BYTE_VALUE_END_OF_STREAM = -1;
+
+	private final InputStream in;
+	private final byte[] variablePrefix;
+	private final Map<String, String> keyValuePairs;
+	private final Queue<Integer> inQueue;
+	private final Queue<Integer> outQueue;
+	private boolean lookingForSuffix = false;
+
+
+	protected ReplacingVariablesInputStream(InputStream in, String variablePrefix, Map<String, String> keyValuePairs) {
```

Comment:
Dit zou alle problemen moeten verhelpen.
```suggestion
	protected ReplacingVariablesInputStream(InputStream in, String variablePrefix, Properties properties) {
```